### PR TITLE
Feature/reject-close-modal

### PR DESCRIPTION
**Pull request by ericjim**
Pass an additional callback to the actionCallback which resets the in progress state of the modal.  Instead of calling `close`, the user of the component can use `rejectClose` if some async actions occurs, fails, and the modal should not be dismissed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Please follow [semantic versioning](http://semver.org/). **Seriously, this is su
 
 Once you've picked a new version number, update the `version` field in `package.json` to that version. This is what NPM will use. Also run this command locally:
 
-`npm run prepublish`
+`npm run prepublishOnly`
```
Comment (SethDavenport) [inline README.md]: :+1: thanks

File: src/layout/modal/ws-modal.service.js [hunk 2]
```diff
@@ -80,6 +80,12 @@ export default function ModalService(
     return $q.when();
   }
 
+  function rejectClose() {
+    $timeout(() => {
+      scope.actionInProgress = false;
+    }, 0);
```
Comment (SethDavenport) [inline src/layout/modal/ws-modal.service.js]: What's the reason for wrapping it in a `$timeout`?

Comment (ericjim) [reply src/layout/modal/ws-modal.service.js]: If I do `scope.actionInProgress = false` the UI will not change, so I have to do `scope.$apply(() => { ... })`. That produces an error telling me that `Error: $apply already in progress`. I can wrap the `$apply` around a `setTimeout`, but `$timeout` does my callback along with the `$apply`.

Comment (SethDavenport) [reply src/layout/modal/ws-modal.service.js]: Technically `$timeout(.., 0)` bumps the execution of `actionInProgress` to the next digest cycle and then triggers the next digest. But OK.

**Pull request merged**